microprofile/telemetry/src/main/java/io/helidon/microprofile/telemetry/HelidonTelemetryContainerFilter.java

```diff
@@ -163,6 +153,47 @@ public void filter(final ContainerRequestContext request, final ContainerRespons
         }
     }
 
+    private String spanName(ContainerRequestContext requestContext) {
+        // According to OpenTelemetry semantic conventions for spans, the span name for a REST endpoint should be
+        //
+        // http-method-name low-cardinality-path
+        //
+        // where a low-cardinality path would be, for example /greet/{name} rather than /greet/Joe, /greet/Dmitry, etc.
+
+        if (spanNameFullUrl) {
+            return requestContext.getMethod() + " " + requestContext.getUriInfo().getAbsolutePath().toString();
+        }
+        ExtendedUriInfo extendedUriInfo = (ExtendedUriInfo) requestContext.getUriInfo();
+
+        // Derive the original path (including path parameters) of the matched resource from the bottom up.
+        StringBuilder derivedPath = new StringBuilder();
+
+        Resource resource = extendedUriInfo.getMatchedModelResource();
+        while (resource != null) {
+            derivedPath.insert(0, resource.getPath());
+            if (!resource.getPath().startsWith("/")) {
+                derivedPath.insert(0, "/");
+            }
+            resource = resource.getParent();
+        }
+
+        derivedPath.insert(0, applicationPath())
+                .insert(0, requestContext.getMethod() + " ");
+        return derivedPath.toString();
+    }
+
+    private String applicationPath() {
+        Application app = Contexts.context()
+                .flatMap(it -> it.get(Application.class))
+                .orElseThrow(() -> new IllegalStateException("Application missing from context"));
+
+        if (app == null) {
+            return "";
+        }
+        ApplicationPath applicationPath = getRealClass(app.getClass()).getAnnotation(ApplicationPath.class);
```

Comment:
Since you've already used things like `ExtendedUriInfo` maybe there's a Jersey class that will give you whatever the `@ApplicationPath` annotation ends up setting indirectly?

Reply:
I didn't see a way to do that, after quite some looking.

If you have a specific suggestion I'd welcome it but I didn't find one. The security code Tomas pointed to uses the same technique to get the `Application`.